tests/epyccel/test_epyccel_sets.py

```diff
@@ -77,3 +77,50 @@ def clear_complex():
     python_result = clear_complex()
     assert python_result == pyccel_result
 
+def test_copy_int(language):
+    def copy_int():
+        se = {1,2,4,5}
+        cop = se.copy()
+        return cop
+    epyccel_copy = epyccel(copy_int, language = language)
+    pyccel_result = epyccel_copy()
+    python_result = copy_int()
+    assert isinstance(python_result, type(pyccel_result))
+    assert python_result == pyccel_result
+
+def test_copy_float(language):
+    def copy_float():
+        se = {5.7, 6.2, 4.3, 9.8}
+        cop = se.copy()
+        return cop
+    epyccel_copy = epyccel(copy_float, language = language)
+    pyccel_result = epyccel_copy()
+    python_result = copy_float()
+    assert isinstance(python_result, type(pyccel_result))
+    assert python_result == pyccel_result
+    assert all(isinstance(elem, type(pyccel_result.pop())) for elem in python_result)
+
+def test_copy_complex(language):
+    def copy_complex():
+        se = {7j, 6j, 9j}
+        cop = se.copy()
+        return cop
+    epyccel_copy = epyccel(copy_complex, language = language)
+    pyccel_result = epyccel_copy()
+    python_result = copy_complex()
+    assert isinstance(python_result, type(pyccel_result))
+    assert python_result == pyccel_result
+    assert all(isinstance(elem, type(pyccel_result.pop())) for elem in python_result)
+
+def test_copy_int(language):
+    def copy_int():
+        se = {56, 23, 98, 3, 6}
+        cop = se.copy()
+        return cop
+    epyccel_copy = epyccel(copy_int, language = language)
+    pyccel_result = epyccel_copy()
+    python_result = copy_int()
+    assert isinstance(python_result, type(pyccel_result))
+    assert python_result == pyccel_result
+    assert all(isinstance(elem, type(pyccel_result.pop())) for elem in python_result)
```

Comment:
There are now 2 `test_copy_int` functions. One at line 80 which doesn't test element types, and this one

Reply:
My bad, Fixed in : ea3e338